fdbserver/workloads/KVStoreTest.actor.cpp

```diff
@@ -252,6 +303,204 @@ struct KVStoreTestWorkload : TestWorkload {
 
 WorkloadFactory<KVStoreTestWorkload> KVStoreTestWorkloadFactory("KVStoreTest");
 
+ACTOR Future<Void> testRangeReadClearRanges(KVStoreTestWorkload* workload, KVTest* ptest) {
+	/*
+	    - This test attempts to simulate a range read with various clear ranges
+	    - First we ask our old range read function to read a range of keys, lets say [a,m]
+	    - Then we randomly pick a subset of these keys, lets say [d,i) and designate this as our "clear range"
+	    - We then feed the range [a, m] to the new range read function and get the result
+	    - We then assert if the result contains all the keys in the range [a, m] except for those in [d, i)
+	*/
+	state KVTest& test = *ptest;
+	state Key key;
+	state int byteLimt = 1 << 30;
+	state int rowLimit = 1000;
```

Comment:
It's probably worth it to randomly (maybe use `BUGGIFY`) set these values to edge cases, like 0.

Reply:
Adding separate test cases like you did below for the clear range at the end is probably the better solution. Being able to parameterize the test code would make this easier.

Reply:
Yeah i think for these (esp since these tests do not run in simulation) its better to not use buggify and test exact cases. If there are weird edge cases im sure simulation would catch them with tests doing range reads